analysis/bulk-deconvolution/exploratory-notebooks/compare-epic-quantiseq.Rmd

```diff
@@ -0,0 +1,441 @@
+      } else {
+        color_map <- color_map_bref
+      }
+      
+       print(make_barplot(project_df, color_map))
+
+    }
+  )
+
+```
+
+Next we'll make a version of these plots which emphasizes shared cell types; cell types that are unique to a reference will be grouped together.
+In addition, we'll also collapse the "Macrophages" cell types in `quanTIseq` into a single "Macrophages" (which `TRef` contains) by summing values for M1 and M2. 
+
+```{r, message = FALSE}
+# sum the quantiseq macrophage to be a single cell type
+quantiseq_macro <- deconv_df |> 
+  dplyr::filter(stringr::str_starts(celltype, "Macrophages.M")) |>
+  dplyr::group_by(project_name, project_id, sample_id, method) |>
+  dplyr::summarize(proportion = sum(proportion)) |>
+  dplyr::ungroup() |>
+  dplyr::mutate(celltype_consensus = "Macrophages")
+
+# update consensus df to have this new macrophage grouping for quantiseq
+consensus_macrophages_df <- consensus_df |>
+  dplyr::filter(
+    !stringr::str_starts(celltype_consensus, "Macrophages.M")
+  ) |>
+  dplyr::bind_rows(quantiseq_macro)
+```
+
+
+```{r, fig.width = 12, fig.height = 6, warning = FALSE}
+# Prepare some plotting colors
+tref_shared <- c("B cells", "CD4 T cells", "CD8 T cells", "NK cells", "Macrophages")
+bref_shared <- c("B cells", "CD4 T cells", "CD8 T cells", "NK cells", "Monocytes", "Neutrophils")
+other_celltypes_label <- "All other reference-specific cell types"
+other_colors <- c( 
+  "Other" = "grey95",
+  "All other reference-specific cell types" = "grey75"
+)
+
+tref_shared_color_map <- c(other_colors, color_map_tref[tref_shared])
+bref_shared_color_map <- c(other_colors, color_map_bref[bref_shared])
+ 
+
+# plot each project
+project_ids |> 
+  purrr::walk(
+    \(pid) {
+            
+      # determine which cell types to plot
+      if (pid %in% tref_projects) {
+        plot_celltypes <- tref_shared
+        color_map <- tref_shared_color_map
+      } else {
+        plot_celltypes <- bref_shared
+        color_map <- bref_shared_color_map
+      }
+  
+      # Subset to only this project and update celltypes to be:
+      # - In the shared vector
+      # - "Other"
+      # - not other but unshared
+      project_df <- consensus_macrophages_df |>
+        dplyr::filter(project_id == pid) |>
+        tidyr::drop_na() |>
+        dplyr::mutate(
+          celltype_grouped = ifelse(
+            celltype_consensus %in% c("Other", plot_celltypes), 
+            celltype_consensus,
+            other_celltypes_label
+          ),
+          celltype_grouped = forcats::fct_relevel(celltype_grouped, names(color_map)),
+          celltype_grouped = forcats::fct_rev(celltype_grouped)
+        ) 
+        
+      
+      print(make_barplot(project_df, color_map, cell_type_column = celltype_grouped))
+      
+    }
+  )
+```
+
+
+### Conclusions
+
+
+* Across all inferences from both methods, B cells are fairly uncommon, but `quanTIseq` generally estimates higher fractions compared to `EPIC`
```

Comment:
I think I might also note the overall differences in T cell populations. It looks like EPIC calls a lot of T cells and virtually no NK cells, while quanTIseq has a lot more NK cells and smaller proportions of T cells (probably due to the difference in granularity between immune cell types present in the different references).  

It looks like you mention it for some specific projects below, but I think the T cell differences also apply to the gliomas.